### PR TITLE
#19646 Avoid Out of Memory with huge file

### PR DESCRIPTION
**Pull request by freddyucv**
Before all the files was store in memory before be added in the tar.gzip, so if the file was huge then you got a Out of Memory

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/publishing/output/TarGzipBundleOutput.java [hunk 1]
```diff
@@ -79,6 +80,20 @@ public void setLastModified(String myFile, long timeInMillis){
 
     }
 
+    public void innerCopyFile(final File source, final String destinationPath) throws IOException {
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/publishing/output/TarGzipBundleOutput.java:83]: Javadoc

File: dotCMS/src/main/java/com/dotcms/publishing/output/BundleOutput.java [hunk 2]
```diff
@@ -55,15 +55,14 @@ public void copyFile(File source, String destinationPath) throws IOException {
...
-                FileUtil.copyFile(source, outputStream);
-            } catch(IOException e) {
-                Logger.error(FileUtil.class, e);
-                throw e;
-            }
+            innerCopyFile(source, destinationPath);
         }
     }
 
+    protected void innerCopyFile(final File source, final String destinationPath) throws IOException {
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/publishing/output/BundleOutput.java:62]: Javadoc

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/publishing/output/TarGzipBundleOutput.java (hunk 1 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/publishing/output/TarGzipBundleOutput.java:83]: @Override annotation

Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/publishing/output/TarGzipBundleOutput.java:84]: done https://github.com/dotCMS/core/pull/20366/commits/7987a2ab154551b9e6b63a477d7028df144bcb29

**Review by dsilvam — APPROVED** [review]

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

**Review by jdotcms — APPROVED** [review]

**Pull request closed without merging**